dms/models/dms_file.py

```diff
@@ -387,11 +386,10 @@ def _compute_extension(self):
     @api.depends("name", "content")
     def _compute_mimetype(self):
         for record in self:
-            mimetype = record.name and mimetypes.guess_type(record.name)[0]
-            if not mimetype and record.content:
-                binary = base64.b64decode(record.with_context({}).content or "")
-                mimetype = guess_mimetype(binary, default="application/octet-stream")
-            record.res_mimetype = mimetype
+            binary = base64.b64decode(record.with_context({}).content or "")
+            record.res_mimetype = guess_mimetype(
+                binary, default="application/octet-stream"
+            )
```

Comment:
Perhaps another possible solution (taking as a reference what odoo does in `ir.attachment`) is something similar to this:

```suggestion
            mimetype = record.name and mimetypes.guess_type(record.name)[0]
            if not mimetype or mimetype == 'application/octet-stream':
                mimetype = guess_mimetype(base64.b64decode(record.content))                
            record.res_mimetype = mimetype
```